src/app/Http/Controllers/Operations/ListOperation.php

```diff
@@ -73,30 +73,41 @@ public function search()
 
         $this->crud->applyUnappliedFilters();
 
-        $totalRows = $this->crud->model->count();
-        $filteredRows = $this->crud->query->toBase()->getCountForPagination();
-        $startIndex = request()->input('start') ?: 0;
+        $start = (int) request()->input('start');
+        $length = (int) request()->input('length');
+        $search = request()->input('search');
+
         // if a search term was present
-        if (request()->input('search') && request()->input('search')['value']) {
+        if ($search && $search['value'] ?? false) {
             // filter the results accordingly
-            $this->crud->applySearchTerm(request()->input('search')['value']);
-            // recalculate the number of filtered rows
-            $filteredRows = $this->crud->count();
+            $this->crud->applySearchTerm($search['value']);
         }
         // start the results according to the datatables pagination
-        if (request()->input('start')) {
-            $this->crud->skip((int) request()->input('start'));
+        if ($start) {
+            $this->crud->skip($start);
         }
         // limit the number of results according to the datatables pagination
-        if (request()->input('length')) {
-            $this->crud->take((int) request()->input('length'));
+        if ($length) {
+            $this->crud->take($length);
         }
         // overwrite any order set in the setup() method with the datatables order
         $this->crud->applyDatatableOrder();
 
         $entries = $this->crud->getEntries();
 
-        return $this->crud->getEntriesAsJsonForDatatables($entries, $totalRows, $filteredRows, $startIndex);
+        // if show entry count is disabled we use the "simplePagination" technique to move between pages.
+        if ($this->crud->getOperationSetting('showEntryCount')) {
+            $totalEntryCount = (int) (request()->get('totalEntryCount') ?? $this->crud->getTotalQueryCount());
```

Comment:
@pxpm  we need to check if the value is empty instead of null only in case of disabling ConvertEmptyStringsToNull